packages/devextreme/testing/testcafe/tests/scheduler/viewOffset/multiCellSelection.ts

```diff
@@ -8,88 +9,96 @@ fixture.disablePageReloads`Offset: Multi cell selection`
 
 const SCHEDULER_SELECTOR = '#container';
 
+// TODO Vinogradov: Change it to a general dispose widget helper function.
+const disposeScheduler = ClientFunction(() => {
+  const scheduler = ($(SCHEDULER_SELECTOR) as any).dxScheduler('instance');
+
+  scheduler?.dispose();
+}, { dependencies: { SCHEDULER_SELECTOR } });
+
 [
   0,
   -120,
   120,
 ].forEach((offset) => {
-  [
-    { view: 'day', dragOptions: { from: [0, 0], to: [5, 0] } },
-    { view: 'week', dragOptions: { from: [2, 2], to: [3, 3] } },
-    { view: 'month', dragOptions: { from: [2, 2], to: [3, 1] } },
-    { view: 'timelineDay', dragOptions: { from: [0, 0], to: [0, 3] } },
-    { view: 'timelineMonth', dragOptions: { from: [0, 0], to: [0, 3] } },
-  ].forEach(({ view, dragOptions }) => {
-    test(`Multi cell selection should work (view: ${view}, offset: ${offset})`, async (t) => {
+  [true, false].forEach((rtlEnabled) => {
+    [
+      { view: 'day', dragOptions: { direction: 'increase', from: [0, 0], to: [7, 0] } },
+      { view: 'day', dragOptions: { direction: 'decrease', from: [7, 0], to: [0, 0] } },
+      { view: 'week', dragOptions: { direction: 'increase_0', from: [0, 2], to: [7, 2] } },
+      { view: 'week', dragOptions: { direction: 'decrease_0', from: [7, 2], to: [0, 2] } },
+      { view: 'week', dragOptions: { direction: 'increase_1', from: [1, 3], to: [8, 4] } },
+      { view: 'week', dragOptions: { direction: 'decrease_1', from: [8, 4], to: [1, 3] } },
+      { view: 'week', dragOptions: { direction: 'increase_2', from: [6, 3], to: [6, 5] } },
+      { view: 'week', dragOptions: { direction: 'decrease_2', from: [6, 5], to: [6, 3] } },
+      { view: 'week', dragOptions: { direction: 'increase_3', from: [0, 0], to: [11, 6] } },
+      { view: 'week', dragOptions: { direction: 'decrease_3', from: [11, 6], to: [0, 0] } },
+      { view: 'month', dragOptions: { direction: 'increase', from: [2, 2], to: [3, 1] } },
+      { view: 'month', dragOptions: { direction: 'decrease', from: [3, 1], to: [2, 2] } },
+      { view: 'timelineDay', dragOptions: { direction: 'increase', from: [0, 0], to: [0, 5] } },
+      { view: 'timelineDay', dragOptions: { direction: 'decrease', from: [0, 5], to: [0, 0] } },
+      { view: 'timelineMonth', dragOptions: { direction: 'increase', from: [0, 0], to: [0, 3] } },
+      { view: 'timelineMonth', dragOptions: { direction: 'decrease', from: [0, 3], to: [0, 0] } },
+    ].forEach(({ view, dragOptions }) => {
```

Comment:
I would recommend to use matrix option generator instead of nested loops. They are hard to read

Reply:
Discussed and decided to replace loops with option matrix generator.